internal/buildinformation/build_info.go

```diff
@@ -0,0 +1,83 @@
+package buildinformation
+
+import (
+	"runtime/debug"
+
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/errors"
+)
+
+const (
+	goarch      string = "GOARCH"
+	goos        string = "GOOS"
+	vcstype     string = "vcs"
+	vcsrevision string = "vcs.revision"
+	vcstime     string = "vcs.time"
+)
+
+type BuildInfo struct {
+	commitSha       string
+	architecture    string
+	vcsTime         string
+	vcsType         string
+	operatingSystem string
+	goVersion       string
+}
+
+// GetBuildInfo returns the current Build Information as a BuildInfo type (struct) using a built-in debug package and its ReadBuildInfo() function.
+// Upon error, an empty struct is returned along with a logged error.
+func GetBuildInfo() (*BuildInfo, *errors.ServiceError) {
+	b := &BuildInfo{}
+	info, ok := debug.ReadBuildInfo()
+	if !ok {
+		return nil, errors.GeneralError("unable to get build info")
+	}
+	b.goVersion = info.GoVersion
+	for _, i := range info.Settings {
+		switch i.Key {
+		case vcstime:
+			b.vcsTime = i.Value
+		case goarch:
+			b.architecture = i.Value
+		case vcstype:
+			b.vcsType = i.Value
+		case goos:
+			b.operatingSystem = i.Value
+		case vcsrevision:
+			b.commitSha = i.Value
+		}
+	}
+	return b, nil
+}
+
+// GetCommitSha returns the current commitSha (version).
+// The returned string can be empty if binary was not built using VCS.
+func (b *BuildInfo) GetCommitSha() string {
```

Comment:
`Sha` should be capitalized to `SHA`. Acronyms should be capitalized as long as don't affect exportability. The same should be done for the attribute in the struct